src/browserExt/mimeTypeHandler.js

```diff
@@ -0,0 +1,181 @@
+	it under the terms of the GNU Affero General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+	
+	Zotero is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU Affero General Public License for more details.
+
+	You should have received a copy of the GNU Affero General Public License
+	along with Zotero.  If not, see <http://www.gnu.org/licenses/>.
+	
+	***** END LICENSE BLOCK *****
+*/
+
+/**
+ * Intercepts web requests and offers to import Zotero recognised resources into Standalone.
+ * 
+ * Includes CSL styles, RIS, Bibtex and Refer files
+ */
+
+(function() {
+
+"use strict";
+
+const url = require('url');
+
+Zotero.MIMETypeHandler = {
+	cslMIMETypes: new Set(["text/x-csl", "application/vnd.citationstyles.style+xml"]),
+	importMIMETypes: new Set([
+		"application/x-endnote-refer", "application/x-research-info-systems",
+		"application/x-inst-for-scientific-info",
+		"text/x-bibtex", "application/x-bibtex",
+		// Non-standard
+		"text/x-research-info-systems",
+		"text/application/x-research-info-systems", // Nature serves this
+		"text/ris", // Cell serves this
+		"ris" // Not even trying
+	]),
+	ignoreURL: new Set(),
+
+	enable: function() {
+		Zotero.WebRequestIntercept.addListener('headersReceived', Zotero.MIMETypeHandler.observe);
+	},
+	
+	disable: function() {
+		Zotero.WebRequestIntercept.removeListener('headersReceived', Zotero.MIMETypeHandler.observe);
+	},
+
+	observe: function(details) {
+		if (!details.responseHeadersObject['content-type']) return;
+		if (Zotero.MIMETypeHandler.ignoreURL.has(details.url)) {
+			// Ignore url when user cancels 
+			Zotero.MIMETypeHandler.ignoreURL.delete(details.url);
+			return;
+		}
+		
+		let URI = url.parse(details.url);
+		let contentType = details.responseHeadersObject['content-type'].split(';')[0];
+		if (Zotero.MIMETypeHandler.cslMIMETypes.has(contentType)) {
+			Zotero.MIMETypeHandler.confirm(details, `Would you like Zotero to install a style from '${details.url}'?`)
+				.then(function(response) {
+					if (response) {
+						Zotero.MIMETypeHandler.importFile(details, 'csl');
+					}
+				}
+			);
+			// Don't continue until we get confirmation
+			return {redirectUrl: 'javascript:'};
+		} else if (Zotero.Prefs.get('interceptAndImport') && 
+			Zotero.MIMETypeHandler.importMIMETypes.has(contentType)) {
+			
+			let hosts = Zotero.Prefs.get('allowedInterceptHosts');
+			let isEnabledHost = hosts.indexOf(URI.host) != -1;
+			if (isEnabledHost) {
+				Zotero.MIMETypeHandler.importFile(details, 'import');
+			} else {
+				Zotero.MIMETypeHandler.confirm(details, `Would you like to use Zotero to import items from ${URI.host}?\n\n` +
+					'You can manage automatic file import in Zotero Connector preferences.').then(function(response) {
```

Comment:
Yea i'll just get a react component and we will inject over a page. Seems like this is inevitable (although kind of awful to do), given the types of interactions Zotero offers. Popovers not usable for prompts, as they cannot be opened programatically.